src/audio/chain_dma.c

```diff
@@ -121,9 +121,28 @@ static int chain_host_stop(struct comp_dev *dev)
 	int err;
 
 	err = dma_stop(cd->chan_host->dma->z_dev, cd->chan_host->index);
-	if (err < 0)
+	if (err < 0) {
+		if (err == -EBUSY) {
+			int retries = 100;
+
+			comp_warn(dev, "dma_stop() fail chan_index = %u, retrying...",
+				  cd->chan_link->index);
+
+			/*
+			 * FIXME: ugly workaround for
+			 * https://github.com/thesofproject/sof/issues/8686
+			 */
+			while (err == -EBUSY && --retries) {
+				k_busy_wait(10);
+				err = dma_stop(cd->chan_link->dma->z_dev, cd->chan_link->index);
```

Comment:
Actually, @ujfalusi noted this is likely not doing actual stop from second iterations onwards, so this is a no-op operation (with the HD-DMA driver). Let me revise the workaround to be more explicit...

Reply:
I guess this would be identical and more explicit:
```diff
diff --git a/src/audio/chain_dma.c b/src/audio/chain_dma.c
index 696973146404..cea2a2e5c8f3 100644
--- a/src/audio/chain_dma.c
+++ b/src/audio/chain_dma.c
@@ -121,7 +121,7 @@ static int chain_host_stop(struct comp_dev *dev)
 	int err;
 
 	err = dma_stop(cd->chan_host->dma->z_dev, cd->chan_host->index);
-	if (err < 0)
+	if (err < 0 && err != -EBUSY)
 		return err;
 
 	comp_info(dev, "chain_host_stop(): dma_stop() host chan_index = %u",
```
With added comment